src/nus/cs2103/addressbook/AddressBook.java

```diff
@@ -30,188 +39,258 @@
-     * Commands using an index to target a person will refer to the indices in this list.
-     *
-     * Persons in this list are the same objects as the corresponding elements in the backing model.
-     * Deleting persons in the backing model may not necessarily update this latest view snapshot.
+     * Stores the most recent list of persons shown to the user as a result of a user command.
+     * This is a subset of the full list. Deleting persons in the pull list does not delete
+     * those persons from this list.
      */
     private static List<String[]> latestPersonListingView = getAllPersonsInAddressBook(); // initial view is of all
-    private static String latestUserInput;
 
-    private static String storageFilePath;
+    /**
+     * Latest command entered by the user.
+     */
+    private static String userCommand;
 
-    private static boolean isExitRequested = false;
+    /**
+     * The path to the file used for storing person data.
+     */
+    private static String storageFilePath;
 
     /*
      * ==============NOTE TO STUDENTS======================================
-     * Notice how this method solves the whole problem at a very high level. We
-     * can understand the high-level logic of the program by reading this method
-     * alone.
+     * Notice how this method solves the whole problem at a very high level.
+     * We can understand the high-level logic of the program by reading this
+     * method alone.
      * ====================================================================
      */
     public static void main(String[] args) {
+        showWelcomeMessage();
         processProgramArgs(args);
-        showToUser(MESSAGE_WELCOME);
-        loadAddressBookModelFromStorage();
-        while (!isExitRequested) {
-            latestUserInput = getUserInput();
-            String feedback = parseAndExecuteCommand(latestUserInput);
+        loadDataFromStorage();
+        while (true) {
+            userCommand = getUserInput();
+            echoUserCommand(userCommand);
+            String feedback = executeCommand(userCommand);
             showToUser(feedback);
         }
-        cleanup();
+    }
+
+    /*
+     * ==============NOTE TO STUDENTS======================================
+     * The method header comment can be omitted if the method is trivial
+     * and the header comment is going to be almost identical to the method
+     * signature anyway.
+     * ====================================================================
+     */
+    private static void showWelcomeMessage() {
+        showToUser(MESSAGE_WELCOME);
+    }
+
+    /*
+     * ==============NOTE TO STUDENTS======================================
+     * Parameter description can be omitted from the method header comment
+     * if the parameter name is self-explanatory.
+     * In the method below, '@param userInput' comment has been omitted.
+     * ====================================================================
+     */
```

Comment:
How about simple pure functions (no side effects)? Like getters `getX()` and predicates `isX(testee)`. The header comment can be merged into the `@return` tag body.